receiver/promtailreceiver/promtail.go

```diff
@@ -0,0 +1,130 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package promtailreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/promtailreceiver"
+
+import (
+	"context"
+	"path"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/loki/clients/pkg/promtail/api"
+	"github.com/grafana/loki/clients/pkg/promtail/targets"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/entry"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/promtailreceiver/internal"
+)
+
+const (
+	operatorType = "promtail_input"
+)
+
+func init() {
+	operator.Register(operatorType, func() operator.Builder { return NewConfig() })
```

Comment:
What happens when this is executed twice? If it would fail, then wrap this on a `sync.Once`.

Reply:
init() function is called only once, so `operator.Register` will be called only once as well

Reply:
I checked what `operator.Register` does. It basically fills the `map[string]func() Builder`
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/ff0a5123be74c6e2a59f3631c640e2753261d358/pkg/stanza/operator/registry.go#L35

If `operator.Register` executes twice, the builder function in the map will be overridden. No errors will occur.
I would keep `init()` function as it is because it looks the same way in all stanza receivers. Having `sync.Once` only in one stanza receiver (promtail receiver) will raise questions in the future. 
From my understanding adding `sync.Once` to `init()` function in all stanza receivers is not necessary because executing `operator.Register` twice will not lead to failure.

@djaglowski please correct me if I'm wrong. If we still need `sync.Once` in stanza receivers `init()` function I would do it in separate PR, because this change should be general for all stanza receivers I think

Reply:
@mar4uk, your assessment appears correct to me. You're using the same registration pattern in the `pkg/stanza` operators.